lib/src/component_declaration/builder_helpers.dart

```diff
@@ -120,9 +124,61 @@ abstract class UiProps extends component_base.UiProps with GeneratedClass {
       ' Will be removed in 4.0.0.')
   @toBeGenerated String get propKeyNamespace => throw UngeneratedError(member: #propKeyNamespace);
 
+  @toBeGenerated PropsMetaCollection get $meta => throw UngeneratedError(member: #meta);
+
   @override @toBeGenerated Map get props => throw UngeneratedError(member: #props);
 }
 
+extension UiPropsMeta on UiProps {
+  static const consumedPropsAssertMessage = 'consumedProps must be either an Iterable<PropsMeta> or PropsMeta';
+
+  /// A collection of metadata for the prop fields in all prop mixins used by
+  /// this props instance's generated props class.
+  ///
+  /// Synonymous with [UiComponent2]'s `propsMeta`.
+  ///
+  /// This can be used to derive consumed props by usage in conjunction with [addUnconsumedProps]
+  /// and [addUnconsumedDomProps].
+  PropsMetaCollection get staticMeta => $meta;
+
+
+  /// Copies props from the provided [props] instance to this [UiProps] instance,
+  /// filtering out props found in [consumedProps].
```

Comment:
#nit redundant type reference https://dart.dev/guides/language/effective-dart/documentation#avoid-redundancy-with-the-surrounding-context

and #nit it's a little abiguous where the filtering occurs in this context

```suggestion
  /// Copies key-value pairs from the provided [props] map into this map, 
  /// excluding those with keys found in [consumedProps].
```